src/main/scala/com/gu/mobile/notifications/football/lib/NotificationsClient.scala

```diff
@@ -35,6 +39,24 @@ trait NotificationHttpProvider extends HttpProvider with Logging {
   }
 }
 
-trait NotificationsClient extends ApiClient with NotificationHttpProvider {
-  def host: String = GoalNotificationsConfig.guardianNotificationsHost
-}
\ No newline at end of file
+class NotificationClientException(message: String) extends Exception(message)
+
+trait NotificationsClient {
+  val apiClient :ApiClient
+
```

Comment:
I'm not super happy about this, The only reason I'm replicating the old notification client functionality by throwing an exception on errors is because this application is set up to only retry on exceptions. If I let this return Either[Error,Unit]  there are tests that fail because it takes the Left(Error) response as successful.
Maybe I can do something in Streams.scala line 85 to let ReactiveX know that a left response should be treated as a failure